lib/handler/mruby.c

```diff
@@ -844,17 +873,24 @@ void h2o_mruby_run_fiber(h2o_mruby_context_t *ctx, mrb_value receiver, mrb_value
             run_again = 1;
         } else {
             h2o_mruby_callback_t callback = ctx->shared->callbacks.entries[callback_index];
-            input = callback(ctx, input, receiver, args, &run_again);
+            input = callback(ctx, input, &receiver, args, &run_again);
         }
         if (mrb->exc != NULL)
             goto GotException;
-        if (run_again == 0)
-            goto Exit;
+        if (run_again == 0) {
+            if (RARRAY_LEN(ctx->resumers) == 0)
+                goto Exit;
+            receiver = mrb_ary_pop(mrb, ctx->resumers);
+        }
 
         mrb_gc_protect(mrb, receiver);
         mrb_gc_protect(mrb, input);
     }
 
+    if (status == 0) {
```

Comment:
Sorry I forgot to remove this check and goto. I introduced _h2o__finish_child_fiber, so this is no longer needed. Could you remove this?